app/src/marketplace/components/Modal/ChooseAccessPeriodDialog/index.jsx

```diff
@@ -80,18 +88,42 @@ export const ChooseAccessPeriodDialog = ({
 
     const isValidTime = useCallback(() => !BN(time).isNaN() && BN(time).isGreaterThan(0), [time])
 
+    const isValidPrice = useCallback(() => {
+        if (paymentCurrency === paymentCurrencies.ETH) {
+            if (BN(priceInEth).isNaN() || !BN(priceInEth).isGreaterThan(0) || !BN(priceInEth).isFinite()) {
+                return false
+            }
+        }
+
+        if (paymentCurrency === paymentCurrencies.DAI) {
+            if (BN(priceInDai).isNaN() || !BN(priceInDai).isGreaterThan(0) || !BN(priceInDai).isFinite()) {
+                return false
+            }
+        }
+
+        return true
+    }, [paymentCurrency, priceInEth, priceInDai])
+
     const getProductPrices = useCallback(async () => {
         if (isValidTime()) {
             setLoading(true)
-            const [ethValue, daiValue] = await getUniswapEquivalents(priceInData)
+
+            const ethValue = await uniswapDATAtoETH(priceInData)
+            const daiValue = await uniswapDATAtoDAI(priceInData)
+
+            let ethUSDEquivalent = await uniswapETHtoDATA(ethValue.toString())
+            ethUSDEquivalent = ethUSDEquivalent.toFixed() / (Number(dataPerUsd) || 1)
```

Comment:
Is `dataPerUsd` already a `BigNumber` here?

Reply:
It comes through as a `NumberString` prop unfortunately.